File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/datasource/FpsDataSource.java
package com.didichuxing.doraemonkit.kit.performance.datasource;

import com.didichuxing.doraemonkit.kit.performance.PerformanceDataManager;
import com.didichuxing.doraemonkit.kit.performance.widget.LineData;

public class FpsDataSource implements IDataSource {
    @Override
    public LineData createData() {
        float rate = PerformanceDataManager.getInstance().getLastFrameRate();
        return LineData.obtain(rate, Math.round(rate) + "");
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/FpsDataSource.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

import com.didichuxing.doraemonkit.kit.performance.manager.PerformanceDataManager
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData.Companion.obtain
import kotlin.math.roundToInt

/**
 *
 * Desc:帧率数据工厂实现类
 * <p>
 * Date: 2020-06-09
 * Updater:
 * Update Time:
 * Update Comments:
 *
 * Author: pengyushan
 */
class FpsDataSource : IDataSource {
    override fun createData(): LineData? {
        val rate: Float = PerformanceDataManager.instance.lastFrameRate
        return obtain(rate, rate.roundToInt().toString() + "")
    }
}